src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -135,45 +135,26 @@ class WorkspaceService(userInfo: UserInfo, dataSource: DataSource, containerDAO:
-        case None =>
-          val bucketName = createBucketName(workspaceRequest.name)
-          Try( gcsDAO.createBucket(userInfo, workspaceRequest.namespace, bucketName) ) match {
-            case Failure(err) =>
-              RequestComplete(StatusCodes.Forbidden,s"Unable to create bucket for ${workspaceRequest.namespace}/${workspaceRequest.name}: "+err.getMessage)
-            case Success(_) =>
-              Try( setupWorkspaceGroupACLs(WorkspaceName(workspaceRequest.namespace, workspaceRequest.name), bucketName) ) match {
-                case Failure(err) =>
-                  gcsDAO.deleteBucket(userInfo, workspaceRequest.namespace, bucketName)
-                  RequestComplete(StatusCodes.Forbidden, s"Unable to create groups for ${workspaceRequest.namespace}/${workspaceRequest.name}: "+err.getMessage)
-                case Success(_) =>
-                  val workspace = Workspace(workspaceRequest.namespace, workspaceRequest.name,bucketName,DateTime.now,userInfo.userEmail,workspaceRequest.attributes)
-                  containerDAO.workspaceDAO.save(workspace, txn)
-                  RequestCompleteWithLocation((StatusCodes.Created,workspace), workspace.toWorkspaceName.path)
-              }
-          }
+      withNewWorkspaceContext(workspaceRequest, txn) { workspaceContext =>
+        RequestCompleteWithLocation((StatusCodes.Created, workspaceContext.workspace), workspaceRequest.toWorkspaceName.path)
       }
     }
 
-  def setupWorkspaceGroupACLs(workspaceName: WorkspaceName, bucketName: String): Unit =
-    dataSource inTransaction { txn =>
-      gcsDAO.setupACL(userInfo, bucketName, workspaceName)
-    }
-
   def getWorkspace(workspaceName: WorkspaceName): PerRequestMessage =
     dataSource inTransaction { txn =>
-      withWorkspaceContextAndPermissions(workspaceName, WorkspaceAccessLevel.Read, txn) { workspaceContext =>
-        val permissionsPair = gcsDAO.getWorkspace(userInfo.userEmail, workspaceName).head
-        val response = WorkspaceListResponse(permissionsPair.accessLevel,
-                            workspaceContext.workspace,
-                            getWorkspaceSubmissionStats(workspaceContext, txn),
-                            gcsDAO.getOwners(permissionsPair.workspaceName))
-        RequestComplete(response)
+      withWorkspaceContext(workspaceName, txn) { workspaceContext =>
+        val accessLevel = gcsDAO.getMaximumAccessLevel(userInfo.userEmail,workspaceContext.workspace.bucketName)
+        if ( accessLevel < WorkspaceAccessLevel.Read )
+          RequestComplete(http.StatusCodes.NotFound, noSuchWorkspaceMessage(workspaceName))
+        else {
+          val response = WorkspaceListResponse(accessLevel,
+            workspaceContext.workspace,
+            getWorkspaceSubmissionStats(workspaceContext, txn),
+            gcsDAO.getOwners(workspaceContext.workspace.bucketName))
+          RequestComplete(response)
+        }
```

Comment:
Because the old code was silly.  It was getting the user's access level (which it had just retrieved and checked, by the way) by getting all groups that the user was a part of and filtering that by workspace.  Now we just get the access level once.